subprojects/integ-tests/src/test/kotlin/org/gradle/kotlin/dsl/integration/DependencyManagementIntegrationTest.kt

```diff
@@ -0,0 +1,125 @@
+
+            dependencies {
+                api("in-block:accessor")
+                api("in-block:accessor-with-action")
+                api("in-block:string-invoke")
+                api("in-block:string-invoke-with-action")
+                api("direct:accessor")
+                api("direct:accessor-with-action")
+                api("direct-block:string-invoke")
+                api("direct-block:string-invoke-with-action")
+            }
+
+            // Declare dependency constraints
+
+            dependencies {
+                constraints {
+                    api("in-block:accessor:1.0")
+                    api("in-block:accessor-with-action") {
+                        version { strictly("1.0") }
+                    }
+                }
+                (constraints) {
+                    "api"("in-block:string-invoke:1.0")
+                    "api"("in-block:string-invoke-with-action") {
+                        version { strictly("1.0") }
+                    }
+                }
+            }
+            dependencies.constraints.api("direct:accessor:1.0")
+            dependencies.constraints.api("direct:accessor-with-action") {
+                version { strictly("1.0") }
+            }
+            (dependencies.constraints) {
+                "api"("direct-block:string-invoke:1.0")
+                "api"("direct-block:string-invoke-with-action") {
+                    version { strictly("1.0") }
+                }
+            }
+
+            repositories {
+                ivy {
+                    url = uri("${existing("repo")}")
+                    layout("pattern") {
+                        this as IvyPatternRepositoryLayout
+                        artifact("[organisation]/[module]-[revision].[ext]")
+                    }
+                }
+            }
+        """)
+
+        build("dependencies", "--configuration", "api").apply {
+            assertThat(output, containsMultiLineString("""
+                api - API dependencies for source set 'main'. (n)
+                +--- in-block:accessor (n)
+                +--- in-block:accessor-with-action (n)
+                +--- in-block:string-invoke (n)
+                +--- in-block:string-invoke-with-action (n)
+                +--- direct:accessor (n)
+                +--- direct:accessor-with-action (n)
+                +--- direct-block:string-invoke (n)
+                \--- direct-block:string-invoke-with-action (n)
+            """))
+        }
+
+        listOf(
+            "in-block:accessor",
+            "in-block:accessor-with-action",
+            "in-block:string-invoke",
+            "in-block:string-invoke-with-action",
+            "direct:accessor",
+            "direct:accessor-with-action",
+            "direct-block:string-invoke",
+            "direct-block:string-invoke-with-action"
+        ).forEach { dep ->
+            build("dependencyInsight", "--configuration", "compileClasspath", "--dependency", dep).apply {
```

Comment:
:x: Have you considered alternatives to avoid running that many builds?

Reply:
The `dependencyInsight` report is the sole one giving the required information and it can only target one dependency at a time. Those builds will reuse the daemon.

Reply:
Fair enough.